helix-term/src/application.rs

```diff
@@ -265,11 +290,56 @@ impl Application {
                     self.jobs.handle_callback(&mut self.editor, &mut self.compositor, callback);
                     self.render();
                 }
-                _ = &mut self.editor.idle_timer => {
-                    // idle timeout
-                    self.editor.clear_idle_timer();
-                    self.handle_idle_timeout();
+                event = self.editor.wait_event() => {
```

Comment:
Does this need bias? Otherwise the event might fill up the other events.

Reply:
I think it's best to keep it biased. I think it's best to prioritize key presses so the user has the best experience, otherwise e.g. if there are a bunch of LSP messages coming through, it could introduce a perceived lag in the responsiveness. Likewise, we don't want to delay file writes because of debugger events or config changes.